runtime-attributes/src/lib.rs

```diff
@@ -1,6 +1,10 @@
 //! Proc Macro attributes for the [Runtime](https://github.com/rustasync/runtime) crate. See the
 //! [Runtime](https://docs.rs/runtime) documentation for more details.
 
+// This file is currently a bit messy because proc macros are not allowed to be split into multiple
+// files. This means that the code for all targets needs to live in this one file. We don't love
+// it, but it beats the alternatives.
```

Comment:
proc-macros are mostly normal crates, the top-level `#[proc_macro_attribute] pub fn` must live in this file, but they can just call into internal functions split across other modules.

Reply:
> but they can just call into internal functions split across other modules.

I believe I tried that, and it didn't work which prompted me to write this comment. Seems like I probably did something wrong.

Reply:
Oh, I probably tried to split the macros into separate modules. You're right (: